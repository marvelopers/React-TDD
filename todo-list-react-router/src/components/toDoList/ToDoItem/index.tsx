import React from 'react';
import styled from 'styled-components';
import Button from '../Button';

interface ToDoItemProps {
  text: string;
  onClick: () => void;
}

const ToDoItem = ({ text, onClick }: ToDoItemProps) => (
  <ToDo onClick={onClick}>
    <div>{text}</div>
    <Button
      text="x"
      onClick={() => {
        console.log('삭제');
      }}
    />
  </ToDo>
);

export default ToDoItem;

const ToDo = styled.li`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 8px 0px;
  border-bottom: 1px solid gray;
`;
