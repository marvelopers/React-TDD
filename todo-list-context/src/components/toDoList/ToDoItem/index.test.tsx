import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';

import ToDoItem from './index';

describe('<ToDoItem/>', () => {
  it('renders component correctly', () => {
    const { container } = render(<ToDoItem text="default text" onClick={() => jest.fn()} />);

    const toDoItem = screen.getByText('default text');
    expect(toDoItem).toBeInTheDocument();

    const deleteButton = screen.getByText('x');
    expect(deleteButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('Clicks the x button', () => {
    const handleClick = jest.fn();
    render(<ToDoItem text="default text" onClick={handleClick} />);

    const deleteButton = screen.getByText('x');
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(deleteButton);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
