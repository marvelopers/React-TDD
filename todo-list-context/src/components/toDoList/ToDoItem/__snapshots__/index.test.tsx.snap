// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToDoItem/> renders component correctly 1`] = `
.c1 {
  padding: 10px 20px;
  background-color: #304FFE;
  border-radius: 8px;
  border: none;
  outline: none;
  text-align: center;
  color: #dbf6fd;
  cursor: pointer;
}

.c1:hover {
  background-color: #1E40FF;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 0px;
  border-bottom: 1px solid gray;
}

<div>
  <li
    class="c0"
  >
    <div>
      default text
    </div>
    <button
      class="c1"
    >
      x
    </button>
  </li>
</div>
`;
